tst/EnergyPlus/unit/Humidifiers.unit.cc

```diff
@@ -4,411 +4,245 @@
+		thisHum.HumType_Code = 2;
+		thisHum.NomCapVol = 4.00E-5;
+		thisHum.NomPower = 103710.0;
+		thisHum.ThermalEffRated = 1.0;
+		thisHum.FanPower = 0.0;
+		thisHum.StandbyPower = 0.0;
+		thisHum.SchedPtr = ScheduleAlwaysOn;
+		thisHum.SchedPtr = ScheduleAlwaysOn;
+
+		FinalSysSizing.allocate( CurSysNum );
+		FinalSysSizing( CurSysNum ).MixTempAtCoolPeak = 20.0;
+		FinalSysSizing( CurSysNum ).MixHumRatAtCoolPeak = 0.00089;
+		FinalSysSizing( CurSysNum ).DesMainVolFlow = 1.60894;
+		FinalSysSizing( CurSysNum ).HeatMixHumRat = 0.05;
+		FinalSysSizing( CurSysNum ).CoolSupHumRat = 0.07;
+		FinalSysSizing( CurSysNum ).HeatSupHumRat = 0.10;
+
+		// resize the humidifier nominal capacity and gas use rate
+		thisHum.NomCapVol = 4.00E-5;
+		thisHum.NomPower = 103710;
+		OutBaroPress = 101325.0;
+		thisHum.SizeHumidifier();
+		EXPECT_DOUBLE_EQ( 0.040000010708118504, thisHum.NomCap );
+		EXPECT_DOUBLE_EQ( 103710.42776358133, thisHum.NomPower );
+
+		// calculate gas use rate and energy at full load
+		thisHum.AirInMassFlowRate = 1.8919;
+		thisHum.AirInTemp = 20.0;
+		thisHum.AirInEnthalpy = 25000.0;
+		thisHum.InletWaterTempOption = 1;
+		thisHum.CurMakeupWaterTemp = 20.0;
+		OutBaroPress = 101325.0;
+
+		thisHum.CalcGasSteamHumidifier( 0.040000010708118504 );
+		EXPECT_DOUBLE_EQ( 103710.42776358133, thisHum.GasUseRate );
+
+		thisHum.ReportHumidifier();
+		EXPECT_DOUBLE_EQ( 93339384.987223208, thisHum.GasUseEnergy );
+	}
+
+	TEST_F( HumidifiersFixture, GetHumidifierInput ) {
+		std::string const idf_objects = delimited_string({
+			"Version,8.3;",
+			"Humidifier:Steam:Gas,",
+			"  Main Gas Humidifier,     !- Name",
+			"  ,                        !- Availability Schedule Name",
+			"  autosize,                !- Rated Capacity {m3/s}",
+			"  autosize,                !- Rated Gas Use Rate {W}",
+			"  0.80,                    !- Thermal Efficiency {-} ",
+			"  ThermalEfficiencyFPLR,   !- Thermal Efficiency Modifier Curve Name",
+			"  0,                       !- Rated Fan Power {W}",
+			"  0,                       !- Auxiliary Electric Power {W}",
+			"  Mixed Air Node 1,        !- Air Inlet Node Name",
+			"  Main Humidifier Outlet Node,  !- Air Outlet Node Name",
+			"  ;                        !- Water Storage Tank Name",
+			"  Curve:Cubic,",
+			"    ThermalEfficiencyFPLR,   !- Name",
+			"    1.0,                     !- Coefficient1 Constant",
+			"    0.0,                     !- Coefficient2 x",
+			"    0.0,                     !- Coefficient3 x**2",
+			"    0.0,                     !- Coefficient4 x**3",
+			"    0.0,                     !- Minimum Value of x",
+			"    1.5,                     !- Maximum Value of x",
+			"    ,                        !- Minimum Curve Output",
+			"    ,                        !- Maximum Curve Output",
+			"    Dimensionless,           !- Input Unit Type for X",
+			"    Dimensionless;           !- Output Unit Type",
+		});
+
+		ASSERT_FALSE( process_idf( idf_objects ) );
+
+		GetHumidifierInput();
```

Comment:
@Nigusse I changed this test to use the new input processing capabilities. The assertions are the same as you had  and the only thing I changed was remove the IDD and IDF data structures creation. Instead it now uses process_idf() and the IDF snippet. Make sure this test is still what you intended, but I think it is.